src/nasher/utils/nwn.nim

```diff
@@ -97,34 +97,86 @@ proc gffConvert*(inFile, outFile, bin, args: string, precision: range[1..32] = 4
   except:
     fatal(fmt"Could not create {outFile}:\n" & getCurrentExceptionMsg())
 
-proc removeUnusedAreas*(dir, bin, args: string) =
-  ## Removes any areas not in ``dir`` from the module.ifo file in ``dir``.
+proc isValid(version: string): bool =
+  # Returns true if the version number is plausible.
+  let decomp = version.split('.')
+
+  if decomp.len < 2 or decomp.len > 4:
+    return false
+  
+  for section in decomp:
+    try:
+      discard section.parseUInt
+      result = true
+    except ValueError:
+      return false
+
+proc updateIfo*(dir, bin, args: string, opts: options.Options, target: options.Target) =
+  ## Updates the areas listing in module.ifo, checks for matching .gic/.git files,          
+  ## and sets module name and min version, if specified
   let
     fileGff = dir / "module.ifo"
     fileJson = fileGff & ".json"
     areas = toSeq(walkFiles(dir / "*.are")).mapIt(it.splitFile.name)
+    gics = toSeq(walkFiles(dir / "*.gic")).mapIt(it.splitFile.name)
+    gits = toSeq(walkFiles(dir / "*.git")).mapIt(it.splitFile.name)
 
   if not existsFile(fileGff):
     return
 
   var
     ifoJson = gffToJson(fileGff, bin, args)
     ifoAreas: seq[JsonNode]
+    unmatchedAreas: seq[string]
 
   let
     entryArea = ifoJson["Mod_Entry_Area"]["value"].getStr
-
+    moduleName =
+      if opts.hasKey("modName"): opts.get("modName")
+      else: target.modName
+    moduleVersion =
+      if opts.hasKey("modMinGameVersion"): opts.get("modMinGameVersion")
+      else: target.modMinGameVersion
+
+  # Area List update
   if entryArea notin areas:
     fatal("This module does not have a valid starting area!")
 
-  for key, value in ifoJson["Mod_Area_list"]["value"].getElems.pairs:
-    let area = value["Area_Name"]["value"].getStr
-    if area in areas:
-      ifoAreas.add(value)
-    else:
-      info("Removing", fmt"unused area {area.escape} from module.ifo")
+  if areas.len > 0:
+    display("Updating", "area list")
+    let plurality = (if areas.len > 1: "s" else: "")
+
+    for area in areas:
+      ifoAreas.add(%* {"__struct_id":6,"Area_Name":{"type":"resref","value":area}})
+
+      if area notin gics or area notin gits:
+        unmatchedAreas.add(area)
+
+    if unmatchedAreas.len > 0:
+      warning("The following do not have matching .gic or .git files and will not be accessible " &
+        "in the toolset: " & unmatchedAreas.join(", "))
 
-  ifoJson["Mod_Area_list"]["value"] = %ifoAreas
+    ifoJson["Mod_Area_list"]["value"] = %ifoAreas
+    success(fmt"area list updated --> {areas.len} area{plurality} listed")
+  
+  # Module Name Update
+  if moduleName.len > 0 and moduleName != ifoJson["Mod_Name"]["value"]["0"].getStr:
+    ifoJson["Mod_Name"]["value"]["0"] = %moduleName
+    success("module name set to " & moduleName)
+
+  # Module Min Game Version Update
+  if moduleVersion.isValid:
```

Comment:
Oddly, that error never showed during testing.  My test cases obviously weren't robust enough.  Addes `elif` to avoid the incorrect error message when no version specified by user.